pkg/handler/iam_handler.go

```diff
@@ -139,12 +140,7 @@ func (h *IAMHandler) handlePolicy(ctx context.Context, p *v1alpha1.ResourcePolic
 			return fmt.Errorf("failed to get IAM policy: %w", err)
 		}
 
-		// TODO (#44): Continue to handle policy and alert updatePolicy error
-		// differently.
-		// Update the policy with new IAM binding additions.
-		if err := updatePolicy(cp, p.Bindings, expiry); err != nil {
-			return fmt.Errorf("failed to update IAM policy: %w", err)
-		}
+		updatePolicy(ctx, cp, p.Bindings, expiry)
```

Comment:
I'm not sure we want to blindly ignore the error here - we should check that the error is specifically the one we want to ignore (either with grpcstatus or checking the error string).

Reply:
In this case we do want to ignore blindly. We don't want to fail just because of cleanup failure. Maybe a comment will help

Reply:
Wouldn't a cleanup failure mean we have overzealous permissions somewhere?

Reply:
This is best effort to clean up old AOD IAM bindings **when applying a new AOD IAM binding**.

As a catch-it-all mechanism, at the moment we plan to do a global scan of AOD bindings near [guardian drift detection](https://github.com/abcxyz/guardian/issues/145) and clean up any findings.

We still plan to do the clean up on AOD PR close. But because we allow multiple iterations in an AOD PR, it's possible one asks for entirely different permission in different iterations. E.g. I can start with a commit to ask for BQ permission then switch over to pubsub permission. When we do the clean up, if we only look at the latest commit, we will only clean up the pubsub permission but not the BQ one. Ideally we should look at all **approved commits** in this PR to clean up. That's much challenging to do.

Reply:
Here is the ongoing PR that is relevant: https://github.com/abcxyz/guardian/pull/148/files

Reply:
Close it for now, please reopen if any concerns.